src/java/nps/id/publicapi/java/client/AppListener.java

```diff
@@ -0,0 +1,149 @@
+import nps.id.publicapi.java.client.connection.subscriptions.exceptions.SubscriptionFailedException;
+import nps.id.publicapi.java.client.connection.subscriptions.requests.SubscriptionRequestBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Service;
+import nps.id.publicapi.java.client.security.options.CredentialsOptions;
+
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.UUID;
+
+@Service
+public class AppListener implements ApplicationListener<ContextRefreshedEvent> {
+    private static final Logger logger = LogManager.getLogger(AppListener.class);
+
+    private final int _demoArea = 2;
+    private final String _clientId = UUID.randomUUID() + "-java-demo-client";
+
+    private final StompClientGenericFactory stompClientGenericFactory;
+
+    private final SubscriptionRequestBuilder subscribeRequestBuilder;
+
+    public AppListener(CredentialsOptions credentialsOptions, StompClientGenericFactory stompClientGenericFactory) {
+        this.stompClientGenericFactory = stompClientGenericFactory;
+        this.subscribeRequestBuilder = SubscriptionRequestBuilder.createBuilder(credentialsOptions.getUserName(), "v1");
+    }
+
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
+        try {
+            var edgeUiClient = CreateClient(WebSocketClientTarget.EDGE);
+            var middlewareUiClient = CreateClient(WebSocketClientTarget.MIDDLEWARE);
+
+            // Delivery areas
+            SubscribeDeliveryAreas(edgeUiClient);
+            SubscribeDeliveryAreas(middlewareUiClient);
+
+            // Contracts
+            SubscribeContracts(edgeUiClient, PublishingMode.CONFLATED);
+            SubscribeContracts(middlewareUiClient, PublishingMode.CONFLATED);
+
+            SubscribeContracts(edgeUiClient, PublishingMode.STREAMING);
+            SubscribeContracts(middlewareUiClient, PublishingMode.STREAMING);
+
+            // Configurations
+            SubscribeConfigurations(edgeUiClient);
+            SubscribeConfigurations(middlewareUiClient);
+
+            // Capacities
+            SubscribeCapacities(edgeUiClient, PublishingMode.CONFLATED);
+            SubscribeCapacities(middlewareUiClient, PublishingMode.CONFLATED);
+
+            SubscribeCapacities(edgeUiClient, PublishingMode.STREAMING);
+            SubscribeCapacities(middlewareUiClient, PublishingMode.STREAMING);
+
+            // Order
+            // We wait some time in hope to get some example contracts and configurations that are needed for preparing example order request
+            Thread.sleep(10000);
+            SendOrderRequest(edgeUiClient);
+            SendOrderRequest(middlewareUiClient);
+
+            System.out.print("Press 'x' key to continue logout, unsubscribe and close. . . ");
```

Comment:
This example is a bit limited comparing to the previous one.
We should be able to:

- Consume execution reports
- Send invalid order
- Refresh token

Basically we can keep existing application and only extend it with second PMD connection example.
There is no strict need in overriding it

Reply:
Added Consume execution reports, sending order request, sending invalid order request, sending order modification request, sending invalid order modification request